SonarQube.TeamBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -63,26 +63,27 @@ public AnalyzerSettings SetupAnalyzers(ISonarQubeServer sqServer, TeamBuildSetti
                 throw new ArgumentNullException("projectKey");
             }
 
-            if (!IsCSharpPluginInstalled(sqServer))
+            AnalyzerSettings analyzerSettings = null;
+            if (IsCSharpPluginInstalled(sqServer))
             {
-                logger.LogDebug(Resources.SLAP_CSharpPluginNotInstalled);
-                return null;
-            }
+                this.server = sqServer;
+                this.settings = teamBuildSettings;
+                this.projectKey = sqProjectKey;
+                this.projectBranch = sqProjectBranch;
 
-            this.server = sqServer;
-            this.settings = teamBuildSettings;
-            this.projectKey = sqProjectKey;
-            this.projectBranch = sqProjectBranch;
-
-            AnalyzerSettings analyzerSettings = null;
+                RoslynExportProfile profile = TryGetRoslynConfigForProject();
+                if (profile != null)
+                {
+                    analyzerSettings = ProcessProfile(profile);
+                }
+                else
+                {
+                    logger.LogDebug(Resources.RAP_CSharpPluginNotInstalled);
```

Comment:
Interesting question. This code is in the pre-processor rather than the bootstrapper so it shouldn't be possible to get this far in normal operation if the C# plugin isn't installed. Anyone using the preprocesser in an unsupported way (i.e. calling it directly) might hit this, but that's not a supported scenario so throwing here doesn't seem unreasonable.
However, the ruleset generator tolerates the plugin being missing so failing here would be inconsistent and a change of behaviour. If we do want to explicitly fail if the plugin isn't installed then it might be better to do so at a higher level i.e. in the preprocessor.